testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java

```diff
@@ -532,6 +540,10 @@ public void getChildren(final String path, final Watcher watcher, final Children
                 if (watcher != null) {
                     watchers.put(path, watcher);
                 }
+            } catch (Throwable ex) {
+                log.error("get children : {} error", path, ex);
+                cb.processResult(KeeperException.Code.SYSTEMERROR.intValue(), path, ctx, null);
+                return;
```

Comment:
Better to move line 551 to `try` block and delete `return 546`

Reply:
Fixed